### PR TITLE
fix size for dim-0

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes: https://github.com/Lightning-AI/lightning-thunder/issues/409

`bool(0)` is False and we end up taking incorrect branch.

TODO: 
* [x] Add test

### CODE REVIEW

File: thunder/torch/__init__.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def is_floating_point(a: TensorLike, /) -> bool:
 
 # Handles the size method
 def size(a: TensorLike, /, dim: None | int = None) -> int | Sequence[int]:
-    if dim:
+    if dim is not None:
```
Comment (mruberry) [inline thunder/torch/__init__.py:176]: Nice explicit test against None

File: thunder/tests/opinfos.py [hunk 2]
```diff
@@ -7970,3 +7970,34 @@ def item_sample_generator(op, device, dtype, requires_grad, **kwargs):
...
+
+    for shape in shapes:
+        t = make_t(shape)
+        yield SampleInput(t)
+
+        for d in range(len(shape)):
+            yield SampleInput(t, d)
+
+
+size_opinfo = OpInfo(
```
Comment (mruberry) [inline thunder/tests/opinfos.py:7997]: Maybe we can put this here:

https://github.com/kshitij12345/lightning-thunder/blob/d30195d85ca6782a58553f0fe8677f930f1a9fa5/thunder/tests/opinfos.py#L565

as a tensor property?

Comment (kshitij12345) [reply thunder/tests/opinfos.py:7997]: Right, I missed that. Thanks!

**Review by mruberry — APPROVED** [review]
